assets/js/loadpoint-settings.ts

```diff
@@ -0,0 +1,130 @@
+import { reactive, watch } from "vue";
+
+export interface LoadpointDisplay {
+  index: number;
+  visible: boolean;
+}
+
+export interface LoadpointDisplayItem extends LoadpointDisplay {
+  title: string;
+  originalData: any;
+}
+
+interface LoadpointSettings {
+  order: number[];
+  visibility: { [key: number]: boolean };
+}
+
+const STORAGE_KEY = "settings_lp_order_visibility";
+
+const getStoredSettings = (): LoadpointSettings => {
+  try {
+    const stored = localStorage.getItem(STORAGE_KEY);
+    if (stored) {
+      return JSON.parse(stored);
+    }
+  } catch (error) {
+    console.warn("Failed to parse stored loadpoint settings:", error);
+  }
+  return { order: [], visibility: {} };
+};
+
+const saveSettings = (settings: LoadpointSettings) => {
+  try {
+    localStorage.setItem(STORAGE_KEY, JSON.stringify(settings));
+  } catch (error) {
+    console.warn("Failed to save loadpoint settings:", error);
+  }
+};
+
+const initialSettings = getStoredSettings();
+
+const loadpointSettings = reactive({
+  order: initialSettings.order,
+  visibility: initialSettings.visibility,
+});
+
+// Watch for changes and persist to localStorage
+watch(
+  loadpointSettings,
+  (newSettings) => {
+    saveSettings({
+      order: newSettings.order,
+      visibility: newSettings.visibility,
+    });
+  },
+  { deep: true }
+);
+
+const initializeLoadpointSettings = (loadpointCount: number) => {
+  // Initialize order if not set or if loadpoint count changed
+  if (loadpointSettings.order.length === 0 || loadpointSettings.order.length !== loadpointCount) {
+    loadpointSettings.order = Array.from({ length: loadpointCount }, (_, i) => i);
+  }
+
+  // Initialize visibility if not set (default to visible)
+  for (let i = 0; i < loadpointCount; i++) {
+    if (!(i in loadpointSettings.visibility)) {
+      loadpointSettings.visibility[i] = true;
+    }
+  }
+};
+
+const getLoadpointOrder = () => loadpointSettings.order;
+
+const setLoadpointOrder = (order: number[]) => {
+  loadpointSettings.order = [...order];
+};
+
+const getLoadpointVisibility = (index: number): boolean => {
+  return loadpointSettings.visibility[index] ?? true;
+};
+
+const setLoadpointVisibility = (index: number, visible: boolean) => {
+  loadpointSettings.visibility[index] = visible;
+};
+
+const getOrderedVisibleLoadpoints = <T>(loadpoints: T[]): Array<T & { id: number }> => {
```

Comment:
Dont understand why we're using generics here. The functions all have loadpoint in its name and are not general purpose. Lets make it concrete input and return types.